root/static/scripts/edit/components/FormRowNameWithGuessCase.js

```diff
@@ -59,14 +63,36 @@ export function runReducer(
 ): void {
   switch (action.type) {
     case 'guess-case': {
+      if (newState.read().valueBeforePreview === null) {
+        newState.set(
+          'field', 'value',
+          GuessCase.entities[action.entity.entityType].guess(
+            newState.read().field.value ?? '',
+          ),
+        );
+      } else {
+        // Updating the name is unneeded if a preview is already shown.
+        newState.set('valueBeforePreview', null);
+      }
+      break;
+    }
+    case 'show-guess-case-preview': {
+      const value = newState.read().field.value ?? '';
+      newState.set('valueBeforePreview', value);
```

Comment:
Just to mention it, I took a different approach here and in the `MB/Control/GuessCase.js` jQuery code than I'd previously taken in `forms.tt` for the tracklist editor. Instead of putting the preview in a parallel `<input>` element and swapping between showing it and the real input, I'm saving the real input's original value and making it display the preview value directly.

I originally went with the separate preview `<input>` elements since it felt safer to not overwrite the real value, but I think that approach will be tough to use in this React component or the jQuery code (where I think I might need to add the preview input to the form, yuck).

I could try to switch the tracklist editor to not use preview `<input>` elements and instead just store the original values in data attributes or something like that, if you'd prefer.